doc/chapters/design/index.rst

```diff
@@ -0,0 +1,39 @@
+SoftwareContainer Design
+************************
+
+SoftwareContainer has been designed as a building block enabling sandboxing of application contents in a larger system.
+
+The SoftwareContainer container achieves this by providing a structured way to limit what IPC mechanims are made available to each application. By targetting the IPC mechanisms rather than the applications or services directly, the sandboxing mechanism can be deployed in a non-intrusive way, i.e. completely transparent to both applications and services.
+
+Each IPC is managed by a *gateway*. The IPC needs of each service is described in *service manifest* files, defining *capabilities* and mapping these to individual *gateway configurations*. This can be individual objects or interfaces on d-bus, ipfilter rules, access to device nodes or something completely different.
+
+Each application is then associated with a set of *capabilities* in an *application manifest*. These are used to configure the sandbox in which the application is then launched.
```

Comment:
We had an offline discussion that we believe "capabilities" are part of the SC domain, and that it is useful to mention the integration pattern that we promote, which contains e.g. "application manifests". For that reason we decided it could be good to try and separate the pure SC domain things and and keep the rest but have it in e.g. a "Integration pattern" section or similar.